typing/jkind.ml

```diff
@@ -1250,49 +1277,65 @@ module Const = struct
         name : string
       }
 
-    let mk_jkind ~mode_crossing ~nullability (layout : Layout.Const.t) =
+    let mk_jkind ~mode_crossing ~nullability ~separability
+        (layout : Layout.Const.t) =
       let mod_bounds =
         (match mode_crossing with
         | true -> Mod_bounds.min
         | false -> Mod_bounds.max)
         |> Mod_bounds.set_nullability nullability
+        |> Mod_bounds.set_separability separability
       in
       { layout; mod_bounds; with_bounds = No_with_bounds }
 
     let any =
-      { jkind = mk_jkind Any ~mode_crossing:false ~nullability:Maybe_null;
+      { jkind =
+          mk_jkind Any ~mode_crossing:false ~nullability:Maybe_null
+            ~separability:Non_separable;
         name = "any"
       }
 
     let any_mod_everything =
-      { jkind = mk_jkind Any ~mode_crossing:true ~nullability:Maybe_null;
+      { jkind =
+          mk_jkind Any ~mode_crossing:true ~nullability:Maybe_null
+            ~separability:Non_separable;
         name = "any mod everything"
       }
 
+    (* CR layouts v3: replace with [any_separable] when or_null arrays are implemented. *)
     let any_non_null =
-      { jkind = mk_jkind Any ~mode_crossing:false ~nullability:Non_null;
+      { jkind =
+          mk_jkind Any ~mode_crossing:false ~nullability:Non_null
+            ~separability:Separable;
         name = "any_non_null"
       }
 
+    (* CR layouts v3: replace with [any_separable] when or_null arrays are implemented. *)
     let any_non_null_mod_everything =
-      { jkind = mk_jkind Any ~mode_crossing:true ~nullability:Non_null;
+      { jkind =
+          mk_jkind Any ~mode_crossing:true ~nullability:Non_null
+            ~separability:Separable;
```

Comment:
This also disagrees with the doc for `everything`

Reply:
Fixed in https://github.com/ocaml-flambda/flambda-backend/pull/3854.